boards/arm/stm32f334_disco/doc/index.rst

```diff
@@ -0,0 +1,128 @@
+.. _stm32f334_disco_board:
+
+ST STM32F334 DISCOVERY
+######################
+
+Overview
+********
+
+The STM32F344DISCOVERY kit features an ARM Cortex-M4 based STM32F334C8
+MCU with everything required for beginners and experienced users to get
+started quickly.
+
+- STM32 microcontroller in LQFP48 package (64K Flash, 16K RAM)
+- Extension header for all I/Os for quick connection to prototyping
+  board and easy probing
+- On-board, ST-LINK/V2 for PCB version A or B or ST-LINK/V2-B for PCB version
+  C and newer, debugger/programmer with SWD connector
+- Board power supply: through USB bus or from an external 3 V or 5 V supply
+  voltage
+- External application power supply: 3 V and 5 V
+- high brightness LED with PWM controlled buck power supply (the STM32F334 part
+  is in fact intended for use in accurate switch mode control applications).
+
+- Six LEDs:
+
+       - 3.3 V power on (LD1)
+       - USB communication (LD2)
+       - Four user LEDs: red, orange, green, bue (LD3, LD4, LD, LD6)
+
+- Two push-buttons: USER and RESET
+- USB USER with Mini-B connector
+
+.. image:: img/stm32f334_disco.jpg
+     :width: 520px
+     :align: center
+     :height: 267px
+     :alt: STM32F344DISCOVERY
+
+More information about the board can be found at the
+`STM32F344DISCOVERY website`_.
+
+Hardware
+********
+
+- STM32F334C8 in LQFP48 package
+- ARM 32 bit Cortex M4 CPU with FPU etc
+- 72MHz max clock frequency
+- 64K Flash, 16K RAM
+- 37 GPIOs
+- 6 timers
```

Comment:
one of the timers is HRTIM, which could be mentioned since it is key feature of the F334 and in this discovery kit